apps/deploy-app.md

```diff
@@ -0,0 +1,252 @@
+
+```
+epw-wind
+│   README.md
+│
+├───.github
+│   └───workflows
+│           ci.yaml
+│
+└───app
+    │   app.py
+    │   Dockerfile
+    │   requirements.txt
+    │
+    └───assets
+            sample.epw
+```
+
+Once done, run the following command from inside the app folder;
+
+```python
+streamlit run app.py
+```
+
+This should render the following in the browser;
+
+![](../.gitbook/assets/pollination-apps/wind_rose.png)
+
+If the app runs as expected, enter "ctrl + c" on the command line to stop the app.
+
+### Step-5: Running the app in a Docker container
+
+Before, deploying the app, it is necessary to run the app in a Docker container. In order to run the app in a Docker container, we will need to install Docker desktop first. Visit this [link](https://www.docker.com/products/docker-desktop) to download the Docker desktop and install it on your system. Once installed, run the docker desktop and keep it running in the background.
+
+Once the docker desktop is running, Let's go to the app folder and use the `run` command from the Pollination-apps library to run the app in a Docker container.
+
+You can find out the structure of this command by using the following command;
+
+```
+pollination-apps run --help
+```
+
+This should show the following in the command line;
+
+```
+Usage: pollination-apps run [OPTIONS] PATH OWNER
+
+  Build and run the application locally.
+
+  Args:     path: Full path to apps folder.     owner: The owner of the app
+  on pollination. e.g. ladybug-tools
+
+Options:
+  -n, --name TEXT  the name of the app (defaults to folder name)
+  -t, --tag TEXT   the tag for this version of the app
+  --help           Show this message and exit.
+```
+
+Let's run the command now to run the app in a Docker container.
+
+```
+pollination-apps run . devang --name epw-wind
+```
+
+This will take a while depending on the size of the app and the number of dependencies in the requirements.txt file. If everything works fine you should see the following appear in the command line;
+
+```
+  You can now view your Streamlit app in your browser.
+
+  Network URL: http://172.17.0.2:8501
+  External URL: http://24.150.248.87:8501
+```
+
+Now, you can view the app in your browser by going to the following URL;
+
+```
+localhost:8501
+```
+
+You should see your app running in the browser. If the app works as expected. Come back to the command line and use the "ctrl + c" command to stop the app.
+
+### Step-6: Generating API token
+
+Once we have tested that the app works inside the Docker container. We are ready to deploy it to Pollination Cloud. To deploy the app, we will login to Pollination cloud first and will create an API token. Go to the "Developer Settings" section of the "Settings" page on your profile and create a new API or retrieve an existing API token. Here, we created a new API token with the name of "epw-wind". We will click on the "refresh" button to get the API token and will copy the API token to the clipboard.
```

Comment:
You should add a note about the API Key being like a password and shouldn't be shared with others or pushed to GitHub. I have already seen at least one example of people doing that.